OZON.ru.js

```diff
@@ -1,604 +1,103 @@
-		var nameXPathRes = doc.evaluate(
-		nameXPath, doc, null, XPathResult.ANY_TYPE, null);
-		var valueXPathRes = doc.evaluate(
-		valueXPath, doc, null, XPathResult.ANY_TYPE, null);
-		if (!IsAudioBook(nameXPathRes, valueXPathRes)) return "book";
-	} else {
-		itemTypeXPath = '//ul[@class="navLine"]/li[2]/a';
-		itemTypeDOMNode =
-		doc.evaluate(itemTypeXPath, doc, null,
-		XPathResult.ANY_TYPE, null).iterateNext();
-		if (itemTypeDOMNode) itemType = itemTypeDOMNode.textContent;
-		else return "";
-		if (itemType == "Цифровые книги") return "book";
-	}
+	var hasIsbn = ZU.xpathText(doc, '//*[@itemprop="isbn"]') !== null;
+	if (hasIsbn) return 'book';
 }
 
-function IsAudioBook(nameXPathRes, valueXPathRes) {
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Формат аудиокниги") return true;
-	}
-	return false;
-}
+function doWeb(doc, url) {
+	var book = new Zotero.Item('book');
 
-function MergeObjects(obj1, obj2) {
-	var result = new Object();
-	for (var attrName in obj1) {
-		result[attrName] = obj1[attrName];
-	}
-	for (var attrName in obj2) {
-		result[attrName] = obj2[attrName];
-	}
-	return result;
-}
+	var content = ZU.xpath(doc, '//div[@class="bContentBlock"]')[0];
 
-function CleanText(text) {
-	var result = text.replace(/^\s*|\s*$/g, '');
-	result = result.replace(/\s+/g, ' ');
-	return result;
-}
+	// basic info
+	book.title = ZU.xpathText(content, './h1[@itemprop="name"]');
+	book.url = url;
 
-function CleanCreator(creatorStr) {
-	return CleanText(creatorStr);
-}
+	book.publisher = ZU.xpathText(content, './/*[@itemprop="publisher"]/a');
+	book.numPages = ZU.xpathText(content, './/*[@itemprop="numberOfPages"]');
 
-function GetCreators(creatorsStr, creatorsCategory) {
-	var result = new Array();
-	var creator;
-	if (creatorsStr.match(",")) {
-		// http://www.ozon.ru/context/detail/id/7341440/
-		var authors = creatorsStr.split(",");
-		for (var i in authors) {
-			creator = CleanCreator(authors[i]);
-			creator = Zotero.Utilities.cleanAuthor(creator, creatorsCategory);
-			result.push(creator);
-		}
-	} else {
-		// http://www.ozon.ru/context/detail/id/7586040/
-		creator = CleanCreator(creatorsStr);
-		creator = Zotero.Utilities.cleanAuthor(creator, creatorsCategory);
-		result.push(creator);
-	}
-	return result;
-}
 
-function GetLanguage(languageStr) {
-	switch (languageStr) {
-	case "Русский":
-		return "ru";
-	case "Английский":
-		return "en";
-	default:
-		return languageStr;
+	var isbn = ZU.xpathText(content, './/*[@itemprop="isbn"]');
+	if (isbn !== null) {
+		book.isbn = /[\d\-]+/.exec(isbn)[0];
+		book.date = /\d{4}(?= г.)/.exec(isbn)[0];
 	}
-}
+	// volume (parsed from the title)
+	var volume = /(т(\.|ом)|vol(\.|ume))\s+\d+/i.exec(book.title);
```

Comment:
That's complicated, because volume number can be in the middle too, like here: http://www.amazon.com/The-Art-Computer-Programming-Searching/dp/0201896850
So I'll have to replace anyway.